nullaway/src/main/java/com/uber/nullaway/fixserialization/Serializer.java

```diff
@@ -224,21 +224,26 @@ public static String serializeSymbol(@Nullable Symbol symbol) {
   }
 
   /**
-   * Serializes the signature of the given {@link Symbol.MethodSymbol} to a string. This code is
-   * based on {@link com.google.errorprone.util.Signatures#prettyMethodSignature}}.
+   * Serializes the signature of the given {@link Symbol.MethodSymbol} to a string.
    *
    * @param methodSymbol The method symbol to serialize.
    * @return The serialized method symbol.
    */
   private static String serializeMethodSignature(Symbol.MethodSymbol methodSymbol) {
     StringBuilder sb = new StringBuilder();
     if (methodSymbol.isConstructor()) {
+      // For constructors, method's simple name is <init> and not the enclosing class, need to
+      // locate the enclosing class.
       Symbol.ClassSymbol encClass = methodSymbol.owner.enclClass();
-      // For constructors, we use the name of the enclosing class.
       Name name = encClass.getSimpleName();
       if (name.isEmpty()) {
-        // For anonymous classes, we use the name of the superclass.
-        name = encClass.getSuperclass().asElement().getSimpleName();
+        // An anonymous class cannot declare its own constructor. Based on this assumption and our
+        // use case, we should not serialize the method signature.
+        throw new RuntimeException(
+            "Did not expect method serialization for anonymous class constructor: "
+                + methodSymbol
+                + ", in anonymous class: "
+                + encClass);
```

Comment:
Can we still add a test case with:

1. A class `Foo` with a constructor taking a non-null arg (i.e. `A(Object o)`)
2. An anonymous subclass of `Foo`, declared as `new Foo(null) { ... }`

Just to make sure it doesn't attempt to serialize an anonymous class constructor when encountering that?

Reply:
Sure, please find it cdbbae48dc056dd53f0b773dabe596b4f2ddab86